client/shaders/nodes_shader/opengl_fragment.glsl

```diff
@@ -93,13 +109,56 @@ void main (void)
 	// Relief mapping
 	if (normalTexturePresent && area_enable_parallax > 0.0) {
 		vec2 ds = eyeRay * PARALLAX_OCCLUSION_SCALE;
+		uv -= ds;
 		float dist = find_intersection(uv, ds);
-		uv += dist * ds;
+		if (texSeamless) {
+			uv += dist * ds;
+		} else if (texTileableVertical == false) {
+			vec2 uv2 = uv + dist * ds;
+			if ((uv.y + uv2.y) < 0.0)
+				uv.y = 0.0;
+			else if ((uv.y + uv2.y) > 1.999)
+				uv.y = 0.999;
+			else
+				uv.y = uv2.y;
+			uv.x = uv2.x;
+		} else {
+			vec2 uv2 = uv + dist * ds;
+			if ((uv.x + uv2.x) < 0.0)
+				uv.x = 0.0;
+			else if ((uv.x + uv2.x) > 1.999)
+				uv.x = 0.999;
+			else
+				uv.x = uv2.x;
+			uv.y = uv2.y;
+		}
+
 #endif
 	} else if (GENERATE_NORMALMAPS == 1 && area_enable_parallax > 0.0) {
 		vec2 ds = eyeRay * PARALLAX_OCCLUSION_SCALE;
+		uv -= ds;
```

Comment:
You should add a comment explaining why you're subtracting this step value from uv whereas this did not happen before.